extensions/guidepost/common/client.mjs

```diff
@@ -0,0 +1,45 @@
+// Copyright 2021 HITCON Online Contributors
+// SPDX-License-Identifier: BSD-2-Clause
+import {MapCoord} from '/static/common/maplib/map.mjs';
+
+const LAYER_GUIDEPOST = {zIndex: -10, layerName: "guidepost"};
+/**
+ * This class is the browser/client side of an extension.
+ * One instance is created for each connected player.
+ */
+class Client {
+  /**
+   * Create the client side of the extension.
+   * @constructor
+   * @param {ClientExtensionHelper} helper - An extension helper object for
+   * servicing various functionalities of the extension.
+   */
+  constructor(helper) {
+    this.helper = helper;
+  }
+
+  /**
+   * Initialize the layer and send the guidepost data as a parameter from json file
+   */
+  async gameStart(){
+    const guideposts = game.assetData = await $.ajax({url: "/static/run/map/watermark/guideposts.json", type: "GET",});
```

Comment:
1. `game` is undefined. I think you copy-pasted it from `client.ejs`.
2. Maybe putting the data under this extension is more reasonable than putting them under `/static/run/map` since this feature is enabled by this extension. Also, the `common` folder in this extension is exported, so moving the data into this extension should be very easy.

Reply:
1. done
2. How to access common from frontend?

Reply:
I thought the guide post location and image would differ from map to map, so it would be more appropriate for it to go in the map's directory? The aim is that if someone wants to use another map, they can just replace the map directory with the new map.